save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt

```diff
@@ -37,12 +37,16 @@ class WarnPlugin(testConfig: TestConfig, testFiles: List<String> = emptyList())
         }
     }
 
-    override fun rawDiscoverTestFiles(resourceDirectories: Sequence<Path>): Sequence<List<Path>> = resourceDirectories
-        .map { directory ->
-            FileSystem.SYSTEM.list(directory)
-                .filter { defaultResourceNamePattern.matches(it.name) }
-        }
-        .filter { it.isNotEmpty() }
+    override fun rawDiscoverTestFiles(resourceDirectories: Sequence<Path>): Sequence<List<Path>> {
+        val warnPluginConfig = testConfig.pluginConfigs.filterIsInstance<WarnPluginConfig>().single()
+        val regex = warnPluginConfig.resourceNamePattern ?: defaultResourceNamePattern
+        return resourceDirectories
+            .map { directory ->
+                FileSystem.SYSTEM.list(directory)
+                    .filter { (regex ?: defaultResourceNamePattern).matches(it.name) }
```

Comment:
It will always match `regex`
```suggestion
                    .filter { (regex).matches(it.name) }
```

Reply:
fixed:)